lib/screens/home/home_screen.dart

```diff
@@ -37,77 +38,181 @@ class _HomeScreenState extends State<HomeScreen> {
-          size: 40,
-        ),
-      ),
-      body: StreamBuilder<List<Wod>>(
-          stream: mainScreenModel.wodStream,
-          builder: (BuildContext context, AsyncSnapshot snapshot) {
-            if (snapshot.hasData) {
-              final List<Wod> wods = snapshot.data;
-              mainScreenModel.groupWods(wods);
-              return Column(
-                children: [
-                  Material(
-                    color: kPrimaryColor,
-                    elevation: 0,
-                    borderRadius: BorderRadius.only(
-                      bottomRight: Radius.circular(10),
-                      bottomLeft: Radius.circular(10),
+
+    return DefaultTabController(
+      length: 2,
+      child: Scaffold(
+          key: _scaffoldKey,
+          appBar: PreferredSize(
+            preferredSize: Size.fromHeight(size.height * 0.42),
+            child: MyAppBar(
+              scaffoldKey: _scaffoldKey,
+              flexibleSpace: FlexibleSpaceBar(
+                centerTitle: true,
+                title: Padding(
+                  padding: const EdgeInsets.all(kDefaultPadding),
+                  child: Center(
+                    child: Column(
+                      mainAxisAlignment: MainAxisAlignment.end,
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: [
+                        Text('Welcome back, Jean!',
+                            style: TextStyle(color: kTextColor)),
+                        DividerBig(),
+                        WodCalendar(),
+                        DividerSmall(),
+                      ],
                     ),
-                    child: Container(
-                      margin: EdgeInsets.only(
-                          left: kDefaultPadding, right: kDefaultPadding),
-                      height: size.height * 0.3,
-                      width: size.width,
-                      decoration: BoxDecoration(
-                        color: kPrimaryColor,
-                        borderRadius: BorderRadius.only(
-                          bottomRight: Radius.circular(10),
-                          bottomLeft: Radius.circular(10),
-                        ),
-                      ),
-                      child: Column(
-                        crossAxisAlignment: CrossAxisAlignment.start,
-                        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                  ),
+                ),
+              ),
+              bottom: TabBar(
+                  indicatorColor: kButtonColor,
+                  tabs: [Tab(text: 'Schedule'), Tab(text: 'WOD')]),
+            ),
+          ),
+          floatingActionButton: FloatingActionButton(
+            onPressed: () =>
+                Navigator.pushNamed(context, AppRoutes.addWodRoute),
+            child: Icon(
+              Icons.add,
+              color: kBackgroundColor,
+              size: 40,
+            ),
+          ),
+          body: TabBarView(
```

Comment:
Please encapsulate each child of `TabBarView` into a different method so it could be more readable